### ISSUE TITLE
Typescript props table overflow

### ISSUE THREAD

**Issue by enagy27** [storybookjs/storybook#10716]
**Describe the bug**
More complex Typescript types such as:
```ts
ComponentClass<ClassAttributes<HTMLInputElement> & InputHTMLAttributes<HTMLInputElement> & { classes?: { ...; } | undefined; }, any> | FunctionComponent<...> | undefined
```
are causing overflow in the props tables.

**To Reproduce**
1. Go to the [Storybook Design System](https://storybook-design-system.netlify.app/?path=/docs/design-system-button--buttons)
2. Replace one of the types provided with the Typescript type above in the inspector.
3. Shrink the window to about 750px wide with the sidebar or about 620px without the sidebar.

**Expected behavior**
Ideally the props table would display a type closer to what is in source. I'm not sure yet if this is something that's fixed with the switch to `react-docgen` 🤞. Regardless my expectation would be that in the worst-case scenario line-breaks would occur on the start of a generic (`<`).

**Screenshots**
<img width="531" alt="Screen Shot 2020-05-09 at 7 07 39 PM" src="https://user-images.githubusercontent.com/13759135/81489379-858ac500-9229-11ea-8c19-8f2ed036b44c.png">

**System:**
I've foregone this section because I'm seeing this behavior in the public-facing docs as well. I'd be happy to provide this, if needed though! Locally I'm running `5.3.18`

**Comment by shilman** [comment]
Thanks for raising this -- definitely not optimal. I'm reconsidering the switch to `react-docgen` since it doesn't support imported types currently and that's a dealbreaker for most "real world" use cases. We need to hustle to get something good in place for 6.0 -- quite a pain in the neck.

**Comment by shilman** [comment]
We've just released zero-config typescript support in 6.0-beta. Please upgrade and test it!

- [Release 6.0.0-beta.12+](https://github.com/storybookjs/storybook/releases/tag/v6.0.0-beta.12)
- [Migration instructions](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#zero-config-typescript)
- [Updated docs](https://github.com/storybookjs/storybook/blob/next/docs/src/pages/configurations/typescript-config/index.md)

Thanks for your help and support getting this stable for release!
